agent/pool/pool.go

```diff
@@ -620,6 +590,17 @@ func (p *ConnPool) rpcInsecure(dc string, addr net.Addr, method string, args int
 	return nil
 }
 
+// BlockableQuery represents a read query which can be blocking or non-blocking.
+// This interface is used to set an appropriate read timeout for rpc connections.
+type BlockableQuery interface {
+	// BlockingTimeout returns duration > 0 if the query is blocking.
+	// Otherwise returns 0 for non-blocking queries.
+	BlockingTimeout(maxQueryTime, defaultQueryTime time.Duration) time.Duration
```

Comment:
I found that narrowly defining this interface was better than giving `RPCInfo` a `Timeout` method and forcing every type to return a timeout (in most cases is rpc_hold_timeout)